### PR TITLE
feat: add `SelectionSortRecursive`

### PR DESCRIPTION
**Pull request by alxkm**
Add SelectionSort using recursion and corresponding Test for it.

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/main/java/com/thealgorithms/sorts/SelectionSortRecursive.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    /**
+     * Recursively sorts the array using selection sort.
+     *
+     * @param array the array to be sorted
+     * @param index the current index to start sorting from
+     * @param <T>   the type of elements in the array (must be Comparable)
+     */
+    private static <T extends Comparable<T>> void recursiveSelectionSort(T[] array, int index) {
+        if (array == null || array.length == 0) {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/sorts/SelectionSortRecursive.java:28]:
```suggestion
        if (array.length == 0) {
```

By the way: the `null` check _should_ happen at the level of `SortAlgorithm`.

File: src/main/java/com/thealgorithms/sorts/SelectionSortRecursive.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            return;
+        }
+        if (index == array.length - 1) {
+            return;
+        }
+
+        // Find the minimum element in the remaining unsorted array
+        int minIndex = findMinIndex(array, index);
```
Comment (vil02) [inline src/main/java/com/thealgorithms/sorts/SelectionSortRecursive.java:36]:
```suggestion
        final int minIndex = findMinIndex(array, index);
```

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/sorts/SelectionSortRecursive.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+     * @return the index of the minimum element
+     */
+    private static <T extends Comparable<T>> int findMinIndex(T[] array, int start) {
+        // Base case: if start is the last index, return start
+        if (start == array.length - 1) {
+            return start;
+        }
+
+        // Recursive call to find the minimum index in the rest of the array
+        int minIndexInRest = findMinIndex(array, start + 1);
```
Comment (vil02) [inline src/main/java/com/thealgorithms/sorts/SelectionSortRecursive.java:62]:
```suggestion
        final int minIndexInRest = findMinIndex(array, start + 1);
```

File: src/main/java/com/thealgorithms/sorts/SelectionSortRecursive.java (hunk 1 above)
Comment (alxkm) [reply src/main/java/com/thealgorithms/sorts/SelectionSortRecursive.java]: Thanks. Yes, in other places this is on level before. Moved there.

**Comment by alxkm** [comment]
Recursive method findMinIndex replaced with iterative. To fix StackOverFlowError on huge arrays for two test cases.

File: src/main/java/com/thealgorithms/sorts/SelectionSortRecursive.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    /**
+     * Sorts an array using recursive selection sort.
+     *
+     * @param array the array to be sorted
+     * @param <T>   the type of elements in the array (must be Comparable)
+     * @return the sorted array
+     */
+    public <T extends Comparable<T>> T[] sort(T[] array) {
+        if (array == null || array.length == 0) {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/sorts/SelectionSortRecursive.java:16]:
```suggestion
        if (array.length == 0) {
```

**Review by vil02 — APPROVED** [review]

**Comment by vil02** [comment]
@alxkm it seems that there is some problem:
https://github.com/TheAlgorithms/Java/actions/runs/9708998343/job/26796827946

**Pull request merged**